### PR TITLE
Copy nested fragments

### LINKED ISSUE #98: Unable to copy a fragment
I'm trying to clone a DS.Model instance. When I use get and set for copying fragments I get the following error:
> "Assertion Failed: Fragments can only belong to one owner, try copying instead"

If I use `Ember.copy` I get this:
> "Assertion Failed: Cannot clone an Ember.Object that does not implement Ember.Copyable"

What is the correct way to achieve what I want?

### PR DESCRIPTION
**Pull request by louy**
Closes #98.

### CODE REVIEW

File: packages/model-fragments/lib/fragments/model.js [hunk 1]
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
...
+      if( attribute.type === 'fragment' ) {
+        if( attribute.kind === 'hasMany' ) {
+          newFragment.set(attribute.name, this.get(attribute.name).map(Ember.copy));
+        } else {
+          newFragment.set(attribute.name, Ember.copy( this.get(attribute.name) ));
+        }
+      } else {
+        newFragment.set(attribute.name, this.get(attribute.name));
+      }
+    }.bind(this));
```
Comment (slindberg) [inline packages/model-fragments/lib/fragments/model.js]: `forEach` accepts a context object as its second argument, I'd prefer that you use that instead.

File: packages/model-fragments/lib/fragments/model.js [hunk 2]
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
...
-    Ember.merge(data, this._data);
-    Ember.merge(data, this._attributes);
-
-    return this.store.createFragment(type, data);
+    
+    var newFragment = store.createFragment(type);
+    get(this.constructor, 'attributes').forEach(function(attribute) {
+      if( attribute.type === 'fragment' ) {
+        if( attribute.kind === 'hasMany' ) {
+          newFragment.set(attribute.name, this.get(attribute.name).map(Ember.copy));
```
Comment (slindberg) [inline packages/model-fragments/lib/fragments/model.js]: Can you use the `get` var here and two places below (like you did on line 175).

File: packages/model-fragments/lib/fragments/model.js [hunk 3]
```diff
@@ -170,13 +170,21 @@ var ModelFragment = CoreModel.extend(Ember.Comparable, Ember.Copyable, {
...
-
-    // TODO: handle copying sub-fragments
-    Ember.merge(data, this._data);
-    Ember.merge(data, this._attributes);
-
-    return this.store.createFragment(type, data);
+    
+    var newFragment = store.createFragment(type);
+    get(this.constructor, 'attributes').forEach(function(attribute) {
+      if( attribute.type === 'fragment' ) {
```
Comment (slindberg) [inline packages/model-fragments/lib/fragments/model.js]: Small nit: can you make the spaces around parens consistent with the rest of the repo? e.g. `if (attribute.type === 'fragment') {`

**Comment by slindberg** [comment]
I really like your approach to this! Could you address my inline comments and rebase? I'd also really really appreciate if you added a test to [packages/model-fragments/tests/integration/fragments/nested_test.js](https://github.com/lytics/ember-data.model-fragments/blob/v0.4.0/packages/model-fragments/tests/integration/fragments/nested_test.js) 😄

**Comment by louy** [comment]
Could you please do those changes for me? Unfortunately, I deleted my local fork and I'm now unable to access these commits. The Ember project I was working on is done now and that's probably why I removed my fork. I forgot that I had this PR open. Sorry about that.

**Comment by louy** [comment]
Just found out that it's possible to checkout a PR. Closing in favor of #123.

**Pull request closed without merging**